cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -152,7 +154,8 @@ data DepResolverParams = DepResolverParams {
        depResolverAvoidReinstalls   :: Bool,
        depResolverShadowPkgs        :: Bool,
        depResolverStrongFlags       :: Bool,
-       depResolverMaxBackjumps      :: Maybe Int
+       depResolverMaxBackjumps      :: Maybe Int,
+       depResolverEnableBackjumping :: EnableBackjumping
```

Comment:
Just trying to understand: by keeping `EnableBackjumping` separate from `MaxBackjumps`, then even with disabled "backjumping", in fact every step up in the tree will still be counted as a backjump, is that right?